internal/konnect/client.go

```diff
@@ -152,6 +159,25 @@ func (c *NodeAPIClient) ListNodes() (*ListNodeResponse, error) {
 	return resp, nil
 }
 
+// ListAllNodes call ListNodes() repeatedly to get all nodes in a runtime group.
+func (c *NodeAPIClient) ListAllNodes() ([]*NodeItem, error) {
+	nodes := []*NodeItem{}
+	pageNum := 0
+	for {
+		resp, err := c.ListNodes(pageNum)
+		if err != nil {
+			return nil, err
+		}
+		nodes = append(nodes, resp.Items...)
+		if resp.Page.NextPageNum == 0 {
+			return nodes, nil
+		}
+		// if konnect returns a non-0 NextPageNum, the node are not all listed
+		// and we should start listing from the returned NextPageNum.
+		pageNum = int(resp.Page.NextPageNum)
+	}
+}
```

Comment:
We could turn this into an iterator I think to prevent loading too many nodes into memory at once. I tried something simple like following:

```suggestion
type NodeItemIteratorWrapper struct {
	NodeItem *NodeItem
	Err      error
}

// ListAllNodes call ListNodes() repeatedly to get all nodes in a runtime group.
func (c *NodeAPIClient) ListAllNodes(ctx context.Context) (<-chan NodeItemIteratorWrapper, error) {
	ch := make(chan NodeItemIteratorWrapper)

	go func() {
		pageNum := 0
		for {
			resp, err := c.ListNodes(pageNum)
			if err != nil {
				select {
				case ch <- NodeItemIteratorWrapper{Err: err}:
				case <-ctx.Done():
				}
				close(ch)
				return
			}

			for _, n := range resp.Items {
				select {
				case ch <- NodeItemIteratorWrapper{NodeItem: n}:
				case <-ctx.Done():
					close(ch)
					return
				}
			}

			if resp.Page.NextPageNum == 0 {
				close(ch)
				return
			}

			// if konnect returns a non-0 NextPageNum, the node are not all listed
			// and we should start listing from the returned NextPageNum.
			pageNum = int(resp.Page.NextPageNum)
		}
	}()

	return ch, nil
}
```

Reply:
It would be too complicated if we use such an iterator to list all nodes. Moreover, when the `listAllNodes` is called, the intention is to get the set of all nodes in runtime group (for further updating/deleting), returning a channel will make such usage also be more complicated. Also, there would not be too many nodes in a runtime group, so that there would not be too much concern about saving nodes in memory.